rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/Doctrine/doctrine_string_target_entity_to_class.php.inc

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\Fixture\Doctrine;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+final class MyClass
+{
+    /**
+     * @ORM\ManyToOne(targetEntity="MyOtherClass")
```

Comment:
MyOtherClass need to exists/autoloaded first before, otherwise, it will a guess that it is a string or class name.

You may need to add the class under `/Source` directory

https://github.com/rectorphp/rector-src/tree/main/rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Source

I am on mobile, you may probably looking for `StringClassNameToClassConstantRector` or improve the `StringClassNameToClassConstantRector` to support inside attribute if not yet applied.

Reply:
Thanks for the help. I targeted the test at `StringClassNameToClassConstantRector`, which seems to be the right rule.

The Problem seems to be that the String targetEntity is a short name without namespace and there seems no easy way to get the FQN of that Class. I found a [gist](https://gist.github.com/Zeronights/7b7d90fcf8d4daf9db0c) that tries to do that. Do you know if similar code exists in rector?

Reply:
You may looking for `UseImportsResolver::resolveForNode()` if you want to try to provide a patch:

https://github.com/rectorphp/rector-src/blob/ae45be1cf3ef4103d529423fcd4d41f2223ddb9a/rules/Naming/Naming/UseImportsResolver.php#L23

Loop and get last part of the namespace use when no alias

Reply:
PTAL. I am not sure how to detect the alias properly. Apart from that, it seems to do the job. :)